packages/rest/src/lib/RequestManager.ts

```diff
@@ -239,13 +244,17 @@ export class RequestManager extends EventEmitter {
 
 			// Attach all files to the request
 			for (const attachment of request.attachments) {
-				formData.append(attachment.fileName, attachment.rawBuffer, attachment.fileName);
+				formData.append(attachment.key ?? attachment.fileName, attachment.rawBuffer, attachment.fileName);
```

Comment:
Are you gonna add the multiple files upload in a diff PR then?

Reply:
Yes, was working through PRs in djs, didn't even see that until after I had this PR up, but it feels better suited to a separate PR.